### DISCUSSION TITLE
How to get buyer-name for FBA orders in reports?

### DISCUSSION THREAD

**Discussion by JohnMirro** [amzn/selling-partner-api-models#3423] [General] [answered]
Many customers have been using our MWS app for years. There was a report like `_GET_AMAZON_FULFILLED_SHIPMENTS_DATA_` , which had information about the buyer- column **buyer-name**.

After migrating to SP-API we use report `GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL`, which does not contain such information.

The documentation says that we can use report `GET_AMAZON_FULFILLED_SHIPMENTS_DATA_INVOICING`, but this report is only available for EU marketplaces.

The customer can participate in both EU and American marketplaces and I can get this list by using "marketplaceParticipations". Which report for North America should I use?

Or maybe there is a trick using RDT to get this data for all marketplaces at once without making extra API requests?

**Comment by keyurk-emipro** [comment, marked as answer]
For North America, use **GET_AMAZON_FULFILLED_SHIPMENTS_DATA_TAX** report type with the RDT token.

**Reply by JohnMirro** [reply]
Thank you very much
